src/modm/board/nucleo_g071rb/board.hpp

```diff
@@ -0,0 +1,165 @@
+{
+
+/// STM32G071RB running at 64MHz generated from the internal 16MHz crystal
+// Dummy clock for devices
+struct systemClock {
+	static constexpr uint32_t Frequency = MHz64;
+	static constexpr uint32_t Ahb = Frequency;
+	static constexpr uint32_t Apb = Frequency;
+
+	static constexpr uint32_t Aes		= Ahb;
+	static constexpr uint32_t Rng		= Ahb;
+	static constexpr uint32_t Crc		= Ahb;
+	static constexpr uint32_t Flash		= Ahb;
+	static constexpr uint32_t Exti		= Ahb;
+	static constexpr uint32_t Rcc		= Ahb;
+	static constexpr uint32_t Dmamux	= Ahb;
+	static constexpr uint32_t Dma		= Ahb;
+
+	static constexpr uint32_t Dbg		= Apb;
+	static constexpr uint32_t Timer17	= Apb;
+	static constexpr uint32_t Timer16	= Apb;
+	static constexpr uint32_t Timer15	= Apb;
+	static constexpr uint32_t Usart1	= Apb;
+	static constexpr uint32_t Spi1		= Apb;
+	static constexpr uint32_t I2s1		= Apb;
+	static constexpr uint32_t Timer1	= Apb;
+	static constexpr uint32_t Adc		= Apb;
+	static constexpr uint32_t Comp		= Apb;
+	static constexpr uint32_t ItLine	= Apb;
+	static constexpr uint32_t VrefBuf	= Apb;
+	static constexpr uint32_t SysCfg	= Apb;
+	static constexpr uint32_t Tamp		= Apb;
+	static constexpr uint32_t Bkp		= Apb;
+	static constexpr uint32_t Ucpd2		= Apb;
+	static constexpr uint32_t Ucpd1		= Apb;
+	static constexpr uint32_t LpTimer2	= Apb;
+	static constexpr uint32_t LpUart1	= Apb;
+	static constexpr uint32_t LpTimer1	= Apb;
+	static constexpr uint32_t HdmiCec	= Apb;
+	static constexpr uint32_t Dac		= Apb;
+	static constexpr uint32_t Pwr		= Apb;
+	static constexpr uint32_t I2c2		= Apb;
+	static constexpr uint32_t I2c1		= Apb;
+	static constexpr uint32_t Usart4	= Apb;
+	static constexpr uint32_t Usart3	= Apb;
+	static constexpr uint32_t Usart2	= Apb;
+	static constexpr uint32_t Spi2		= Apb;
+	static constexpr uint32_t Iwdg		= Apb;
+	static constexpr uint32_t Wwdg		= Apb;
+	static constexpr uint32_t Rtc		= Apb;
+	static constexpr uint32_t Timer14	= Apb;
+	static constexpr uint32_t Timer7	= Apb;
+	static constexpr uint32_t Timer6	= Apb;
+	static constexpr uint32_t Timer3	= Apb;
+	static constexpr uint32_t Timer2	= Apb;
+
+	static bool inline
+	enable()
+	{
+		ClockControl::enableInternalClock();	// 16MHz
+		// (internal clock / 1_M) * 8_N / 2_R = 128MHz / 2 = 64MHz
+		ClockControl::enablePll(
+			ClockControl::PllSource::InternalClock,
+			1,	// Pll_M
+			8,	// Pll_N
+			2	// Pll_R
+		);
+		// set flash latency for 64MHz
+		ClockControl::setFlashLatency(Frequency);
+		// switch system clock to PLL output
+		ClockControl::enableSystemClock(ClockControl::SystemClockSource::Pll);
+		ClockControl::setAhbPrescaler(ClockControl::AhbPrescaler::Div1);
+		ClockControl::setApbPrescaler(ClockControl::ApbPrescaler::Div1);
+		// update frequencies for busy-wait delay functions
+		modm::clock::fcpu     = Frequency;
+		modm::clock::fcpu_kHz = Frequency / 1000;
+		modm::clock::fcpu_MHz = Frequency / 1000000;
+		modm::clock::ns_per_loop = ::round(4000.f / (Frequency / 1000000));
```

Comment:
This needs to be verified, I think the Cortex-M0+ has a shorter than normal pipeline or something.

Reply:
Fixed, Cortex-M0+ is similar to M3.